### ISSUE TITLE
Detailed Transactions report is not displaying details by default. (expand/collapse)

### ISSUE THREAD

**Issue by myosotisgit** [opensourcepos/opensourcepos#2745]
### Background information

- [x] Reporting an issue of an unmodified OSPOS installation
- [x] Checked [open and closed issues](https://github.com/opensourcepos/opensourcepos/issues?utf8=%E2%9C%93&q=is%3Aissue) and no similar issue was already reported (please make sure you searched!)
- [x] Read [README](https://github.com/opensourcepos/opensourcepos/blob/master/README.md), [WHATS_NEW](https://github.com/opensourcepos/opensourcepos/blob/master/WHATS_NEW.txt), [INSTALL.md](https://github.com/opensourcepos/opensourcepos/blob/master/INSTALL.md) and [UPGRADE](https://github.com/opensourcepos/opensourcepos/blob/master/UPGRADE.txt)
- [x] Read the [FAQ](https://github.com/opensourcepos/opensourcepos#faq) for any known install and/or upgrade gotchas (in specific PHP extensions installed)
- [x] Read the [wiki](https://github.com/opensourcepos/opensourcepos/wiki)
- [x] Executed any database upgrade scripts if an upgrade pre 3.0.0 (e.g. database/2.4_to_3.0.sql)
- [x] Aware the installation code is in [bintray](https://bintray.com/jekkos/opensourcepos/opensourcepos/view/files?sort=updated&order=asc#files) (see README), and [GitHub master](https://github.com/opensourcepos/opensourcepos/tree/master) is for [developers only](https://github.com/opensourcepos/opensourcepos/wiki/Development-setup) and therefore not complete nor stable

### Question / New Feature

When i open the Report called "Detailed transactions" the details are hidden/collapsed by default. I can expand each transaction to view the details. If you have a lot of sales per day you need to expand each sale to view the details. That is very time consuming. 
- My suggestion: Add a "expand/collapse" icon next to the columns icon.

When i download the report as a 'pdf' the details for each transaction are also not displayed, just the overview. 

When printing the detailed transaction webpage the table header with columns has a white background. This is very unclear to print...

If you feel this could be added then that would be nice. 

**In the meantime, if anyone knows how to autoexpand" the detailed transaction in the code, let me know!**

**Comment by myosotisgit** [comment]
Im searching through the code to find the "plus" and "minus" expand functionality. But so far i cannot find it. Im guessing i should look in the /application/helpers ?

```
[u8934p6180@web0101 application]$ fgrep -rn "expand" *
controllers/Reports.php:1347:                           $attribute_values = expand_attribute_values($definition_names, $drow);
controllers/Reports.php:1463:                           $attribute_values = expand_attribute_values($definition_names, $drow);
helpers/tabular_helper.php:453: return $columns + expand_attribute_values($definition_names, (array) $item) + $icons;
helpers/tabular_helper.php:548:function expand_attribute_values($definition_names, $row)

```

**Comment by odiea** [comment]
Look under app/views/reports/tabular_details at the bottom you will see. onExpandRow:

**Comment by myosotisgit** [comment]
> Look under app/views/reports/tabular_details at the bottom you will see. onExpandRow:

Found it. If i understand it correctly, the function onExpandRow will show the detailed data for this transaction. This function is found in the minified js file and inside the Bower_compenents.

Now how to trigger it so the details always expand by default ?

**Comment by odiea** [comment]
That's as far as I can help.

**Comment by myosotisgit** [comment]
i found this on the Bootstrap Table website: https://bootstrap-table.com/docs/api/events/#onexpandrow

**onExpandRow**
jQuery Event: expand-row.bs.table
Parameter: index, row, $detail
Detail: Fires when click the detail icon to expand the detail view, the parameters contain:
index: the index of the expanded row.
row: the record corresponding to the expanded row.
$detail: the DOM element of the detail div after current tr element, you can use jQuery methods to custom the detail views.

**Comment by myosotisgit** [comment]
This is some complex Javascript wizardry inside bootstrap-table  js .........
Goes above my head right now

**Comment by daN4cat** [comment]
Have you checked whether there is any property allowing you to have it expanded by default?

**Comment by myosotisgit** [comment]
I understand the code a little bit better (but the core team can correct me).
This app uses the bootstrap-table (with a few more extensions) for bootstrap core.
In bootstrap-table are included the "toggle", "export" and also this onExpandRow functionality.

So the question is where is the jquery that checks the mouse click to expand, set it to true by default.
Or add a jquery to body-onload to expand all....

**Comment by myosotisgit** [comment]
I went to the bootstrap-table github page and asked a similar question. This is an answer i got. Now the real question is how to integrate this into ospos?
https://github.com/wenzhixin/bootstrap-table/issues/4922#issuecomment-594092935

And here is the JSFiddle
https://live.bootstrap-table.com/code/UtechtDustin/2095

**Comment by odiea** [comment]
All I had to do was to add the following after the last function and before the </script> in app/views/reports/tabular_details . Then click on any + sign and all the tables will Expand.  Appears to work good but you have to - each one individually.
```
$(() => {
    const $table = $('table');
    $table.on('post-body.bs.table', (e) => {
      const $trs = $(e.currentTarget).find('tbody tr')
      $.each($trs, function (key, tr) {
      	$table.bootstrapTable('expandAllRows', $(tr).data('index'));  
    	})    	
    })
  }); 
```

**Comment by myosotisgit** [comment]
YES! 
That works beautifully. I dont mind to click the minus multiple times. (or just reload, that also works!!)

I was digging a little deeper into Bootstrap-table. It seems that the included version for bootstrap-table is 1.12.x. Current version (with more options for toggles, etc) is 1.16.x.

But im very happy with the code from Bootstrap-table and the comment by @odiea on where to put the code exactly!!!

Now onto my next quest.  Disable a few columns by default (they are all enabled by default...)
